functions/cmd_volume_get_option.pp

```diff
@@ -0,0 +1,34 @@
+# Create a command string to get option `$opt` from gluster volume `$vol`, and
+# optionally compare it against `$comparison`.
+#
+# @param vol [Gluster::VolumeName] Gluster volume name
+# @param opt [Gluster::OptionName] Gluster volume option name
+# @param comparison [Optional[String]] Optional string to compare the existing
+#   value against
+# @return [String]
+#
+# @example Usage
+#
+#   ```puppet
+#   gluster::cmd_volume_get_option('data', 'nfs.disable', String(true))
+#   ```
+#
+function gluster::cmd_volume_get_option(
+  Gluster::VolumeName   $vol,
+  Gluster::VolumeOption $opt,
+  Optional[Any]         $comparison = undef,
+) {
+  $_cmd = "${::gluster_binary} volume get ${vol} ${opt}"
```

Comment:
please don't add leading `::`. Also please use the new facts hash to access facts:
```suggestion
  $_cmd = "${facts['gluster_binary']} volume get ${vol} ${opt}"
```